doc/tutorials/advanced-ensemble.md

```diff
@@ -0,0 +1,603 @@
+   log-file:
+   "/home/vyzo/src/vyzo/advanced-ensemble-demo/root/log/test/supervisor/server.log"
+   addresses:
+   ((unix: "dellicious" "/tmp/ensemble/test/supervisor.sock"))
+   known-servers:
+   (((registry . /test) (unix: "dellicious" "/tmp/ensemble/test/registry.sock"))))
+  registry:
+  (config:
+   ensemble-server-v0
+   identifier:
+   (registry . /test)
+   role:
+   registry
+   exe:
+   "gerbil"
+   args:
+   ("ensemble" "registry" "(registry . /test)")))
+ roles:
+ ((httpd exe:
+         "gerbil"
+         prefix:
+         ("httpd" "server")
+         policy:
+         restart
+         server-config:
+         (config:
+          ensemble-server-v0
+          env:
+          "www"
+          application:
+          ((httpd config:
+                  httpd-v0
+                  root:
+                  "www"
+                  listen:
+                  ("0.0.0.0:8080")
+                  handlers:
+                  (("/handler" . :demo/handler))
+                  enable-servlets:
+                  #t))))))
+```
+
+As we can see, the only server within the ensemble under the
+supervisor is the automatically spawned registry, which facilitates
+local interactions between servers in the ensemble.
+
+Before we can spawn our httpds, we also need to _upload_ the necessary code
+and content:
+```
+2$ cd site/project/.gerbil/
+2$ tar czvf ../../env-www.tar.gz lib
+
+2$ cd site
+2$ tar czvf fs-www.tar.gz www
+
+2$ gerbil ensemble -G env/local control upload --fs site/fs-www.tar.gz ""
+2$ gerbil ensemble -G env/local control upload --env site/env-www.tar.gz www
+```
+
+And that's it, now we are ready to spawn our httpd servers as workers:
+```
+2$ gerbil ensemble -G env/local control start-workers -d /test/www httpd httpd 2
+2$ gerbil ensemble -G env/local control list-servers --pretty
+(((httpd-0 . /test/www) 2348366 running)
+ ((httpd-1 . /test/www) 2348367 running)
+ ((registry . /test) 2348146 running))
+```
+
+And we can interact with them:
+```
+2$ ps auxw | grep gerbil
+2$ curl http://localhost:8080/
+2$ curl http://localhost:8080/servlets/hello.ss
+2$ curl http://localhost:8080/handler
+```
+
+Finally, we can shutdown the entire ensemble, including the supervisor:
+```
+2$ gerbil ensemble -G env/local control shutdown
+```
+
+## Distributed Ensembles
+
+Naturally, we are not limited to controlling a local ensemble behind a
+supervisor.  We can just as easy spawn a distributed ensemble that
+runs in multiple hosts, all running their own supervisor.
+
+The individual supervisors themselves can be spawned and supervised as
+a `systemd` service.
+
+In this tutorial, we build an ensemble with 3 hosts: 2 hosts serving
+http requests with a (local) httpd ensemble each, and another host
+acting as a load balancer using the example `rlb` program.
+
+See [src/tutorial/advanced-ensemble/rlb](https://github.com/mighty-gerbils/gerbil/tree/master/src/tutorial/advanced-ensemble/rlb) for the `rlb` code.
```

Comment:
404s, probably should point to https://github.com/mighty-gerbils/gerbil/blob/ensemble-domains/src/tutorial/advanced-ensemble/rlb

Reply:
because the pr hasn't merged yet!

It is where the file is going to be.